src/db/file.rs

```diff
@@ -224,7 +218,7 @@ pub fn add_path_into_database<P: AsRef<Path>>(conn: &Connection,
                 }
             }
 
-            file_paths_and_mimes.insert(file_path.clone(), mime.clone());
+            file_paths_and_mimes.insert(file_path.clone(), mime.to_owned());
```

Comment:
We should be able to avoid the clone here

Reply:
I don't think we can. `file_path` is borrowed from `currently_uploading`, which can be reused on a later iteration of the loop. I might be able to make `file_paths_and_mimes` store a &'static str but that would require some refactoring elsewhere.

Reply:
Anyway I don't think it should be a big deal since file paths aren't very long compared to the size of a file contents. It will be dropped at the end of the loop in any case.

Reply:
er, sorry I meant the mime type - it's potentially significant (3 pointers + heap to just two pointers).

Reply:
Still doesn't work.

```
error[E0599]: no method named `to_json` found for type `std::vec::Vec<&str>` in the current scope
   --> src/db/file.rs:286:31
    |
286 |         file_list_json.push(v.to_json());
    |                               ^^^^^^^ method not found in `std::vec::Vec<&str>`
    |
    = note: the method `to_json` exists but the following trait bounds were not satisfied:
            `[&str] : rustc_serialize::json::ToJson`
            `std::vec::Vec<&str> : rustc_serialize::json::ToJson`
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope; perhaps add a `use` for it:
            `use serde_json::value::ToJson;`
```

Reply:
If we're willing to switch this from rustc_serialize to serde_json we could make it work, but I'm not sure I'm comfortable doing that without tests.

Reply:
Gah, okay. Thanks for trying.